velox/expression/CastExpr-inl.h

```diff
@@ -301,6 +301,68 @@ VectorPtr CastExpr::applyDecimalToBooleanCast(
   return result;
 }
 
+template <typename FromNativeType>
+VectorPtr CastExpr::applyDecimalToVarcharCast(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& fromType) {
+  VectorPtr result;
+  context.ensureWritable(rows, VARCHAR(), result);
+  (*result).clearNulls(rows);
+  const auto simpleInput = input.as<SimpleVector<FromNativeType>>();
+  auto [precision, scale] = getDecimalPrecisionScale(*fromType);
+  // Calculate the max total size of the raw string buffer.
+  size_t maxTotalResultBytes = rows.countSelected() * precision;
```

Comment:
> size_t maxTotalResultBytes = rows.countSelected() * precision;

Is this accurate? Don't we need to account for a need to include a '.' and a '-' sign?

Reply:
And don't forget first `0`.
```
presto:default> select cast('-0.00012' as decimal(5, 5));
  _col0
----------
 -0.00012
(1 row)
```

Reply:
@mbasmanova @jinchengchenghh Thank you. Updated.

Reply:
> And don't forget first `0`.
> 
> ```
> presto:default> select cast('-0.00012' as decimal(5, 5));
>   _col0
> ----------
>  -0.00012
> (1 row)
> ```

Please add this one to test, this is a corner case for buffer space.

Reply:
Already covered by similar test.
https://github.com/facebookincubator/velox/pull/6210/files#diff-f41a5cc1aa75bd9ae99c7ed93b7a6fe859f3796183fa45969032d5836b9f19e8R1447-R1448

Reply:
If we revoke `getRawStringBufferWithSpace`, we will allocate 32KB buffer at least, small dataset cannot test if the buffer we allocate is really enough. https://github.com/facebookincubator/velox/blob/main/velox/vector/FlatVector.cpp#L215
But anyway, we can review to guarantee the logic is right.

Reply:
And test with other different string, they may not require the largest buffer size with small precision and small scale, we still cannot cover the varcharSize logic.